### ISSUE TITLE
能否上传最新代码？

### ISSUE THREAD

**Issue by magnum-qin** [ZouWang-spider/TCMT#2]
作者您好，在尝试复现您的代码时，我注意到代码中的目标检测用的是FastRCNN而不是YOLOv5，我想确认一下您最终使用的是YoloV5是吗？能否更新一下代码呢？谢谢！

**Comment by ZouWang-spider** [comment]
不好意，近期事情较多，下面是Yolov5单独的代码，当前已经出现Yolov11，所以你也可以尝试其他版本的
import torch
from PIL import Image, ImageDraw
import matplotlib.pyplot as plt

#安装工具
# pip install torch torchvision
# pip install git+https://github.com/ultralytics/yolov5.git


# 加载预训练的YOLOv5模型
model = torch.hub.load('ultralytics/yolov5:v7.0', 'yolov5s')  # 也可以使用yolov5m，yolov5l或yolov5x


# 加载图像并进行预处理
def load_image(image_path):
    image = Image.open(image_path).convert("RGB")
    return image


# 进行目标检测
def detect_objects(model, image):
    results = model(image)  
    return results


# 可视化检测结果，阈值设置为0.5
def visualize_detection(image, results, threshold=0.5):
    boxes = results.xyxy[0].cpu().numpy() 
    labels = results.pred[0][:, -1].cpu().numpy()  
    scores = results.pred[0][:, 4].cpu().numpy()  
    draw = ImageDraw.Draw(image)

    for i in range(len(boxes)):
        if scores[i] > threshold:
            box = boxes[i]
            label = int(labels[i])
            score = scores[i]

            # 绘制边界框
            draw.rectangle(box, outline="red", width=3)
            draw.text((box[0], box[1]), f"{results.names[label]} {score:.2f}", fill="red")

    plt.imshow(image)
    plt.show()


# 主函数
def main(image_path):
    image = load_image(image_path)
    results = detect_objects(model, image)
    visualize_detection(image, results)


# 示例图片路径
image_path = "images/017.png"  

main(image_path)
